java/code/src/com/suse/manager/webui/utils/salt/custom/HwProfileUpdateSlsResult.java

```diff
@@ -78,41 +82,65 @@
      * @return the grains
      */
     public Map<String, Object> getGrains() {
+        if (grains == null) {
+            LOG.warn("grains value is null");
+            return Collections.emptyMap();
+        }
         return grains.getChanges().getRet();
     }
 
     /**
      * @return information about CPUs
      */
     public Map<String, Object> getCpuInfo() {
+        if (cpuInfo == null) {
+            LOG.warn("cpuInfo value is null");
+            return Collections.emptyMap();
+        }
         return cpuInfo.getChanges().getRet();
     }
 
     /**
      * @return exported contents of the udevdb
      */
     public List<Map<String, Object>> getUdevdb() {
+        if (udevdb == null) {
+            LOG.warn("udevdb value is null");
```

Comment:
Not sure if we log a warning in all cases. Not sure if on all architectures really all values exists.
But we can add that and check the logs if something appears always.

Reply:
I added this check only on the field that should be mandatory ( others are `Optional` ). So yes, we can keep it and if it is shown often we can change the return type in `Optional<List<Map<String, Object>>>`  and remove the warning.